api/app/src/main/kotlin/packit/security/oauth2/OAuth2UserService.kt

```diff
@@ -39,13 +39,14 @@ class OAuth2UserService(
 
         var user = userService.saveUserFromGithub(githubInfo.userName(), githubInfo.displayName(), githubInfo.email())
 
-        val principal = UserPrincipal(
-            user.username,
-            user.displayName,
-            roleService.getGrantedAuthorities(user.roles),
-            oAuth2User.attributes
+        return PackitOAuth2User(
+            UserPrincipal(
+                user.username,
+                user.displayName,
+                mutableSetOf(),
```

Comment:
Does this need to be mutable? Could we change `UserPrincipal` to have non-mutable fields?

```kt
val authorities: Collection<GrantedAuthority>,
val attributes: Map<String, Any>
```

The `OAuth2User` interface that `PackitOAuth2User` implements only needs non-mutable accessors.

While we're at it, I don't think we use the attributes anywhere, so maybe we can just remove that field.